actionview/test/template/form_helper/form_with_test.rb

```diff
@@ -381,6 +381,26 @@ def test_form_with
     assert_dom_equal expected, output_buffer
   end
 
+  def test_form_with_with_persisted_to_model
+    post_form = OpenStruct.new(to_model: @post)
+
+    form_with(model: post_form) { }
+
+    expected = whole_form("/posts/123", method: :post) { "" }
```

Comment:
I understand that this was the behaviour before, but it seems wrong to me. Can we make it consistent across helpers?

Reply:
@gmcgibbon I had originally made the behavior consistent, but it felt too disruptive of a change, and deserved a separate PR and discussion.

Is a separate PR a waste of time? If we feel like it is, I can change this diff to consistently infer the `method:` in both `form_for` and `form_with`.

Reply:
👍 Let's do it in another PR. Seems fine to have a separate changelog entry for this change in behaviour anyway.

Reply:
@gmcgibbon I've opened https://github.com/rails/rails/pull/44468.